### DISCUSSION TITLE
cy.intercept() and cy.visit()

### DISCUSSION THREAD

**Discussion by willowloveforever** [cypress-io/cypress#28171] [Questions and Help]
Hello, I am writing login logic using cypress. I used cy.request in beforeEach to obtain the token, and used cy.intercept to place the token in the authorization attribute of the header of all interfaces in test. test. com/api/* */*. Then I use cy.visit in it to request that the homepage product. product page has a default request to obtain data and load the product page. But these requests were all reported incorrectly. I checked that the header in Matched 'cy. intercept()' has authorization. But there is no authorization in the actual request headers. May I ask what I should do. thanks.

**Comment by portyu9** [comment]
Hi @willowloveforever 

What you are seeing in Chrome DevTools is expected.

When you add or change a request header inside `cy.intercept()`, Cypress modifies the request **after it has left the browser**. Because of that, the modified header will not appear in the browser's Network tab.

You can verify what Cypress actually sends by asserting on the intercepted request instead:

```js
cy.intercept('**/api/**', (req) => {
  req.headers.authorization = `Bearer ${token}`
}).as('apiRequest')

cy.visit('/')

cy.wait('@apiRequest')
  .its('request.headers.authorization')
  .should('eq', `Bearer ${token}`)
```

Make sure the intercept is registered before `cy.visit()` causes the application's requests.

If this assertion passes but the server still rejects the request, then inspect the server response/server logs and confirm the exact expected `Authorization` format (for example, whether it requires the `Bearer ` prefix). The absence of the header in Chrome DevTools by itself does **not** mean `cy.intercept()` failed to add it.

Please mark this answer as accepted if it helped, thank you.
